stdlib/public/core/NativeDictionary.swift

```diff
@@ -460,6 +460,28 @@ internal func KEY_TYPE_OF_DICTIONARY_VIOLATES_HASHABLE_REQUIREMENTS(
 }
 
 extension _NativeDictionary { // Insertions
+  /// Insert a new element into uniquely held storage.  Storage must be uniquely
+  /// referenced with adequate capacity.  If `allowingDuplicates` is false,
+  /// `element` must not be already present in the dictionary.
+  @_alwaysEmitIntoClient @inlinable // Introduced in 5.1
```

Comment:
Does this get specialized enough that the unused code path gets DCE'd if the type is known to the client? It'd be a shame to inline stuff they don't need.

Reply:
The flag gets folded into a constant, but the branch on it doesn't get eliminated unless the `insertWithGuaranteedCapacity` call is inlined -- unfortunately it normally isn't. 

In the latest commit, I reorganized the code to move the branch out of the loop, ensuring dead code elimination will trigger.